controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -92,20 +124,296 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
+		switch r.storageClassClaim.Spec.Type {
+		case "blockpool":
+			storageClassClaimStorageType = FulfillStorageClassClaimRequestBlockpool
+		case "sharedfilesystem":
+			storageClassClaimStorageType = FulfillStorageClassClaimRequestSharedfilesystem
+		}
+
+		// Call the `FulfillStorageClassClaim` service on the provider server with StorageClassClaim as a request message.
+		_, err = providerClient.FulfillStorageClassClaim(
+			r.ctx,
+			r.storageCluster.Status.ExternalStorage.ConsumerID,
+			r.storageClassClaim.Name,
+			r.storageClassClaim.Spec.EncryptionMethod,
+			storageClassClaimStorageType,
+		)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to initiate fulfillment of StorageClassClaim: %v", err)
+		}
+
+		// Configuration phase.
+		r.storageClassClaim.Status.Phase = ocsv1alpha1.StorageClassClaimConfiguring
+
+		// Call the `GetStorageClassClaimConfig` service on the provider server with StorageClassClaim as a request message.
+		response, err := providerClient.GetStorageClassClaimConfig(
+			r.ctx,
+			r.storageCluster.Status.ExternalStorage.ConsumerID,
+			r.storageClassClaim.Name,
+		)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to get StorageClassClaim config: %v", err)
+		}
+		resources := response.ExternalResource
+		if resources == nil {
+			return reconcile.Result{}, fmt.Errorf("no configuration data recieved")
+		}
+
+		// Go over the received objects and operate on them accordingly.
+		for _, resource := range resources {
+			data := map[string]string{}
+			err = json.Unmarshal(resource.Data, &data)
+			if err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed to unmarshal StorageClassClaim configuration response: %v", err)
+			}
+
+			switch resource.Kind {
+			case "Secret":
+				secret := &corev1.Secret{}
+				secret.Name = resource.Name
+				secret.Namespace = r.storageClassClaim.Namespace
+				_, err = controllerutil.CreateOrUpdate(r.ctx, r.Client, secret, func() error {
+					err := r.own(secret)
+					if err != nil {
+						return fmt.Errorf("failed to own Secret: %v", err)
+					}
+					m := map[string][]byte{}
+					for k, v := range data {
+						m[k] = []byte(v)
+					}
+					secret.Data = m
+					return nil
+				})
+				if err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to create or update secret %v: %s", secret, err)
+				}
+			case "StorageClass":
+				var storageClass *storagev1.StorageClass
+				if resource.Name == "cephfs" {
+					storageClass = r.getCephFSStorageClass(data)
+				} else if resource.Name == "ceph-rbd" {
+					storageClass = r.getCephRBDStorageClass(data)
+				}
+				err = r.createOrReplaceStorageClass(storageClass)
+				if err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to create or update StorageClass: %s", err)
+				}
+			}
+		}
+
+		// Readiness phase.
+		// Update the StorageClassClaim status.
+		r.storageClassClaim.Status.Phase = ocsv1alpha1.StorageClassClaimReady
```

Comment:
I see you have updated this value at multiple places, but never actually called `r.Client.Status.Update` to update status in server.